tests/components/blebox/test_config_flow.py

```diff
@@ -190,3 +194,107 @@ async def test_async_remove_entry(hass, valid_feature_mock):
 
     assert hass.config_entries.async_entries() == []
     assert config.state is config_entries.ConfigEntryState.NOT_LOADED
+
+
+async def test_flow_with_zeroconf(hass):
+    """Test setup from zeroconf discovery."""
+    result = await hass.config_entries.flow.async_init(
+        config_flow.DOMAIN,
+        context={"source": config_entries.SOURCE_ZEROCONF},
+        data=zeroconf.ZeroconfServiceInfo(
+            host="172.100.123.4",
+            addresses=["172.100.123.4"],
+            port=80,
+            hostname="bbx-bbtest123456.local.",
+            type="_bbxsrv._tcp.local.",
+            name="bbx-bbtest123456._bbxsrv._tcp.local.",
+            properties={"_raw": {}},
+        ),
+    )
+
+    assert result["type"] == FlowResultType.FORM
+
+    with patch("homeassistant.components.blebox.async_setup_entry", return_value=True):
+        result2 = await hass.config_entries.flow.async_configure(result["flow_id"], {})
+        await hass.async_block_till_done()
+
+    assert result2["type"] == FlowResultType.CREATE_ENTRY
+    assert result2["data"] == {"host": "172.100.123.4", "port": 80}
+
+
+async def test_flow_with_zeroconf_when_already_configured(hass):
+    """Test behaviour if device already configured."""
+    entry = MockConfigEntry(
+        domain=config_flow.DOMAIN,
+        data={CONF_IP_ADDRESS: "172.100.123.4"},
+        unique_id="abcd0123ef5678",
+    )
+    entry.add_to_hass(hass)
+    feature: AsyncMock = mock_feature(
+        "sensors",
+        blebox_uniapi.sensor.Temperature,
+    )
+    with patch("blebox_uniapi.box.Box.async_from_host", return_value=feature.product):
```

Comment:
```suggestion
    with patch("homeassistant.components.blebox.config_flow.Box.async_from_host", return_value=feature.product):
```

Usually we only patch the place where its imported instead of globally

Reply:
Suggestion commited. Thanks, I'll keep that in mind next time.